Assets/Scripts/UI/IntroText.cs

```diff
@@ -6,17 +6,52 @@ public class IntroText : MonoBehaviour
 {
     [SerializeField] private DialogueTrigger dialogueTrigger;
     [SerializeField] private DialogueManager dialogueManager;
+    private bool started = false;
 
-    void Start()
+    void Awake()
     {
-        UIEventListener.Instance.PauseGame();
         dialogueTrigger.TriggerDialogue();
     }
 
+    void Start() {
+        UIEventListener.Instance.PauseGame();
+        EnableControls(false);
+    }
+
     void Update()
     {
-        if (!dialogueManager.isDialoging) {
-           UIEventListener.Instance.UnpauseGame(); 
+        Proceed();
+    }
+
+    void Proceed() {
+        if (!dialogueManager || !dialogueManager.isDialoging) return; // Skip if we're done intro-ing
+
+        if (dialogueManager.finalDialogue) {
+            PlayerInput.Controls.Player.Enable(); // Enable controls so we can go in future
+            return; // To stop anything else from triggering the next sentence
+        }
+
+        if (Input.anyKeyDown){
```

Comment:
I would use `InputSystem.onAnyButtonPress` so that we're using the new Unity input system consistently.

Reply:
From what I understand doing this would add a behaviour that happens on all button presses forever? https://docs.unity3d.com/Packages/com.unity.inputsystem@1.2/api/UnityEngine.InputSystem.InputSystem.html

I tried to set it up s.t. the Input.anyKeyDown code is usually not even reached unless dialogueManager is in the middle of dialoging. In my next PR i do want to look into this more though, because I'm sure you're right and there's a better way I can set this up to use the new input system efficiently

Reply:
Yeah good point. I think there's a way to remove the listener when dialoging is done, but it doesn't look super simple based on the docs. Definitely a low priority fix.